### PR TITLE
Profile generalization: cleanup

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR removes some temporary/no longer needed code that was used during profile migration. There's some more cleanup work I'd like to do after this (primarily around loading the CSV file in the specs) but I'd rather do that in a separate PR.

## Related Tickets & Documents

#12161

## QA Instructions, Screenshots, Recordings

Again, please test as much as you can. 

* Onboarding flows
* Profile updates
* General usage of the site (article browsing, etc.)

### UI accessibility concerns?

None

## Added/updated tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: this is primarily a refactoring/cleanup

### CODE REVIEW

File: app/assets/stylesheets/views/comments.scss
Comment (citizen428) [inline app/assets/stylesheets/views/comments.scss:276]: Automatic update from commit hook.

File: app/views/users/_metadata.html.erb
Comment (citizen428) [inline app/views/users/_metadata.html.erb:9]: Automatic update from commit hook.

File: babel.config.js
Comment (citizen428) [inline babel.config.js:58]: Automatic update from commit hook.

File: docs/installation/windows.md
Comment (citizen428) [inline docs/installation/windows.md:293]: Automatic update from commit hook.

File: docs/self-hosting.md
Comment (citizen428) [inline docs/self-hosting.md:7]: Automatic update from commit hook.

File: spec/models/navigation_link_spec.rb
Comment (citizen428) [inline spec/models/navigation_link_spec.rb:16]: Automatic update from commit hook.

**Comment by citizen428** [comment]
Sorry, I had added some reviewers manually and then marking the PR as ready for review added a whole bunch more, I removed some again.

File: app/assets/images/forem-background.svg
Comment (rhymes) [reply app/assets/images/forem-background.svg:2]: Yes, it's set to prettify SVGs

File: app/controllers/api/v0/organizations_controller.rb
Comment (rhymes) [inline app/controllers/api/v0/organizations_controller.rb:36]: no N+1 :D

File: app/controllers/api/v0/users_controller.rb
Comment (rhymes) [inline app/controllers/api/v0/users_controller.rb:13]: I'm starting to think if we should consider (not in this PR), eager loading profiles using `strict_loading: true` see https://blog.saeloun.com/2020/02/25/rails-strict-loading-mode-to-fix-n-1.html - we would need to understand if the profile is always or very often needed when we load the user object

File: db/migrate/20210309064924_remove_profile_columns_from_user.rb
Comment (rhymes) [inline db/migrate/20210309064924_remove_profile_columns_from_user.rb:1]: YOLO 🤣

**Review by rhymes — APPROVED** [review]
Great stuff @citizen428!

File: app/controllers/api/v0/users_controller.rb
Comment (citizen428) [reply app/controllers/api/v0/users_controller.rb:13]: Yes, I have a comment to this effect somewhere. Not explicitly about strict loading, but about figuring out the answer to this issue so we can decide on the best strategy :-)

File: db/migrate/20210309064924_remove_profile_columns_from_user.rb
Comment (citizen428) [reply db/migrate/20210309064924_remove_profile_columns_from_user.rb:1]: Haha, luckily we removed all their usages in the previous PR which didn't blow up too many things. Well, at least not things we noticed 🤣

**Review by juliannatetreault — APPROVED** [review]
I took these changes for a spin locally and didn't notice any new bugs or breakage! 🚀

File: db/migrate/20210309064924_remove_profile_columns_from_user.rb
Comment (djuber) [reply db/migrate/20210309064924_remove_profile_columns_from_user.rb:1]: this is the exciting culmination of a long process. I'm so very happy to see all these `alter table, drop column` commands.

File: docs/frontend/preact.md
Comment (djuber) [inline docs/frontend/preact.md:59]: was this a linter change?

**Review by djuber — APPROVED** [review]

**Review by maestromac — APPROVED** [review]
tested. LGTM!

File: docs/frontend/preact.md
Comment (citizen428) [reply docs/frontend/preact.md:59]: Yes. There were quite a few of them in this PR, looks like lately some things slipped into `main`.

**Comment by citizen428** [comment]
Note to @maestromac and @Zhao-Andy: I didn't want to merge this when nobody with access to fix potential problems is awake, it's not urgent enough to warrant potentially waking someone up over 😃 I therefore tagged this "merge by any core" and leave it up to one of you to merge either today or early next week if you don't want to push out a change like this on a Friday.

**Pull request merged**
